### PR TITLE
SDIT-1223: ✨ Copy across all changes from template typescript project

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: server/app.ts [hunk 1]
```diff
@@ -3,56 +3,39 @@ import express from 'express'
...
   const app = express()
+
   app.set('json spaces', 2)
   app.set('trust proxy', true)
   app.set('port', process.env.PORT || 3000)
 
-  app.use((req, res, next) => {
-    res.locals = {
-      ...res.locals,
-      currentUrlPath: req.baseUrl + req.path,
```
Comment (petergphillips) [inline server/app.ts:42]: this never appeared to be used, so removed

File: server/config.ts [hunk 2]
```diff
@@ -109,9 +118,6 @@ export default {
       enabled: get('FRONTEND_COMPONENTS_ENABLED', 'false') === 'true',
     },
   },
-  analytics: {
-    tagManagerContainerId: get('TAG_MANAGER_CONTAINER_ID', ''),
```
Comment (petergphillips) [inline server/config.ts:113]: ripped out as we're not meant to be using google analytics these days

File: server/data/prisonApiClient.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ export default class PrisonApiClient {
     this.restClient = new RestClient('Prison API', config.apis.prison, token)
   }
 
-  async getPrisonerImage(prisonerNumber: string): Promise<Readable> {
+  getPrisonerImage(prisonerNumber: string): Promise<Readable> {
```
Comment (petergphillips) [inline server/data/prisonApiClient.ts:31]: fixed intellij warning

File: server/middleware/authorisationMiddleware.test.ts [hunk 4]
```diff
@@ -28,44 +28,54 @@ describe('authorisationMiddleware', () => {
...
       redirect: jest.fn(),
+      render: jest.fn(),
     } as unknown as Response
   }
 
   beforeEach(() => {
     jest.resetAllMocks()
   })
 
-  it('should return next when no required roles', async () => {
```
Comment (petergphillips) [inline server/middleware/authorisationMiddleware.test.ts:38]: fixed inteliij warnings about unnecessary `await`

File: server/services/hospitalSearchService.test.ts [hunk 5]
```diff
@@ -12,10 +13,8 @@ jest.mock('../data/prisonApiClient', () =>
 
 const user = {
   username: 'user1',
-  name: 'User',
-  activeCaseLoadId: 'MDI',
   token: 'token-1',
-} as User
+} as Context
```
Comment (petergphillips) [inline server/services/hospitalSearchService.test.ts:17]: copied nomis sync dashboard approach of passing round `context` rather than the `user`

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**